unit_testing/cli/test_cli_sct_denoising_onlm.py

```diff
@@ -0,0 +1,14 @@
+import pytest
+import logging
+
+from spinalcordtoolbox.scripts import sct_denoising_onlm
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.sct_testing
+@pytest.mark.usefixtures("run_in_sct_testing_data_dir")
+def test_sct_denoising_onlm_no_checks():
+    """Run the CLI script without checking results.
+    TODO: Check the results. (This test replaces the 'sct_testing' test, which did not implement any checks.)"""
+    sct_denoising_onlm.main(argv=['-i', 't2/t2.nii.gz', '-v', '2'])
```

Comment:
Hm, this test keeps hanging on specifically Catalina. At first I thought it was a Travis glitch, but it's happened multiple times now. (e.g. [Job #15237.3](https://travis-ci.com/github/neuropoly/spinalcordtoolbox/jobs/488252879#L1266), [Job #15202.3](https://travis-ci.com/github/neuropoly/spinalcordtoolbox/jobs/486386514#L1266).

So, this may actually be an issue with the underlying script + Catalina.

Reply:
i'm catalina (still)-- will check in ~15min

Reply:
hum, no problem on my end:
https://gist.github.com/jcohenadad/82df099e957005a28b32bb96700f91ed

Reply:
CI issues are... the least fun issues to debug. My suggestion would probably be increasing granularity of the logging temporarily (or add more debug print out/checks for that specific function being pytested) to at least start narrow down the region where the hanging happens on CI and potentially rewrite/reimplement those regions if this is consistently giving issues.

My colleague once had a CI issue on that was not reproducible on any local machines and it turned out most likely to be a low level error with the way we update pointers to items within a list dynamically while trying to be efficient. We were only able to narrow down the cause to a region of code and rewrote it but never fully 100% sure what was the precise cause. It costed us about 3 months of off-and-on part-time trying to identify the cause. I really hope what you see now is much easier to debug...

Reply:
also: it is not worth spending hours on this since `sct_denoising_onlm` is probably rarely used

Reply:
> also: it is not worth spending hours on this since sct_denoising_onlm is probably rarely used

:+1:

> My suggestion would probably be increasing granularity of the logging temporarily (or add more debug print out/checks for that specific function being pytested)

As an aside, this is a helpful observation, and another motivation for #2853. Right now, I don't believe SCT has an easy way to change verbosity on the fly during testing -- the `-v` argument has to be specified individually for each and every `main` call.

(Also, use of verbosity in SCT is a bit complicated right now, and hard to summarize. I tried in https://github.com/neuropoly/spinalcordtoolbox/pull/2897#issue-489598310, but that summary could use cleaning up.)